### PR TITLE
Refactor to use the sdc_aws_utils python library and remove redundent code (v2)

### PR DESCRIPTION
**Pull request by dbarrous**
This PR updates the processing lambda function to use the new sdc_aws_utils (https://github.com/HERMES-SOC/sdc_aws_utils), which removes a lot of redundant tasks across the repos into one Python library that can be easily updated from one location. This includes logging, an aws timestream database, and sending Slack notifications. Simplifying the main `lambda_function.py` and moving that logic into `file_processor.py` file.

What changes:
- Switches to one consolidated logging initiated and configured in sdc_aws_utils
- Switches to use the consolidated Slack messaging, which will be updated to the new format
- Switches to use the consolidated timestream database logging
- Moves code from handler to main `file_processor.py`

**Two PRs additional PRs will follow from re-writing and breaking apart the stale PR**

### CODE REVIEW

File: lambda_function/src/file_processor/file_processor.py [hunk 1]
```diff
@@ -194,35 +227,55 @@ def _process_file(self) -> None:
...
+                                    slack_channel=self.slack_channel,
+                                    path=new_file_path,
+                                    alert_type="processed",
+                                )
+                        except Exception as e:
+                            log.error(
+                                {
+                                    "status": "ERROR",
+                                    "message": f"Error when logging to Slack: {e}",
+                                }
```
Comment (Alrobbertz) [inline lambda_function/src/file_processor/file_processor.py:257]: Do you need another try/catch here if there is already one in the utils function for `send_pipeline_notification`?

File: lambda_function/src/file_processor/file_processor.py [hunk 2]
```diff
@@ -194,35 +227,55 @@ def _process_file(self) -> None:
...
-                            action_type="PUT",
-                            file_key=self.file_key,
-                            new_file_key=new_file_key,
-                            source_bucket=destination_bucket,
-                            destination_bucket=destination_bucket,
-                        )
+                        try:
+                            if self.timestream_client:
+                                # Log to timeseries database
+                                log_to_timestream(
```
Comment (Alrobbertz) [inline lambda_function/src/file_processor/file_processor.py:263]: same here, this already has a try/catch inside of it. Do you want to catch the error there or here?

**Review by Alrobbertz — COMMENTED** [review]
Looks great!

File: lambda_function/src/file_processor/file_processor.py (hunk 1 above)
Comment (dbarrous) [reply lambda_function/src/file_processor/file_processor.py]: Great point; I've fixed this and removed the redundant try and catches from this function.

File: lambda_function/src/file_processor/file_processor.py (hunk 2 above)
Comment (dbarrous) [reply lambda_function/src/file_processor/file_processor.py]: ^^

**Review by Alrobbertz — APPROVED** [review]

**Pull request merged**